### ISSUE TITLE
Default link tool not working in Safari

### ISSUE THREAD

**Issue by MrMooky** [codex-team/editor.js#2499]
The default link tool is not working in Safari, even in your demo. Clicking the link icon does not trigger anything.

<img width="283" alt="Screenshot 2023-10-02 at 14 54 56" src="https://github.com/codex-team/editor.js/assets/3603346/98089e90-a1a8-45a7-8ad5-38e74523c9d4">

When editing an existing link (set prior to the OS update), the link field shows and I can edit it.

Device, Browser, OS:
macOS Sonoma (14.0), Safari 17.0

Editor.js version:
Latest available

**Comment by neSpecc** [comment]
Can't reproduce right away. But haven't tested in Sonoma yet. Will check.

**Comment by ChARCOOL** [comment]
Looks like this happens in latest version of MacOS Sonoma (14) and Safari.

**Comment by suleymanbariseser** [comment]
The issue is related the anchorElement. It seems when user focus on input, Safari removes the selection so anchorElement becomes null but Chrome stores input focus as a selection. 

![Safari screenshot](https://github.com/codex-team/editor.js/assets/50797736/75453e6f-368f-468b-bae4-aea58d6204ab)
![Chrome screenshot](https://github.com/codex-team/editor.js/assets/50797736/5fc786d4-7258-4659-9ec6-1010fa84eba5)


Due to focusedElement(anchorElement) does not exist on Safari, inline toolbar closes
 
```ts
if (!focusedElement) {
  /**
   * If there is no selected range, close inline toolbar
   *
   * @todo Make this method more straightforward
   */
  if (!Selection.range) {
    this.Editor.InlineToolbar.close();
  }

  return;
}
```
[link to code](https://github.com/codex-team/editor.js/blob/c9014e670d89f94e0361440b777cc7bb99022655/src/components/modules/ui.ts#L805)

**Comment by collimarco** [comment]
I confirm this bug on Safari 17 MacOS Sonoma. You can also reproduce this issue on your website demo.

**Comment by MrMooky** [comment]
It worked again for me on Version 17.1 (19616.2.9.11.7).

**Comment by ncer** [comment]
I confirm this bug again with Editor.js 2.30.2 on MacOS Sonoma 14.5, Safari 17.5. You can also reproduce this issue again on your website demo.

Previous latest minor version of Editor.js (2.29.1) works fine.

**Comment by philippbuerger** [comment]
Any updates here? Facing the same issue in safari. Other browser working fine

**Comment by neSpecc** [comment]
Resolved by #2864, available in 2.30.7
